src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -121,6 +135,175 @@ public static SyncerFactory of(
         .whenComplete((syncResult, throwable) -> closeClients());
   }
 
+  @Nonnull
+  CompletableFuture<Void> syncMultipleResources(
```

Comment:
by looking at this new method and the existing `sync` method, now it seems `sync` method is pointless. This new method will also work if there is only one option available. Isn't it better if you move this method content into the existing `sync` method and handle options accordingly. wdyt?

Reply:
Yes Jude, But I feel syncMultipleResources method has many checks and time taking and it is unnecessary for a single resource sync. I think I have discussed with the dev team already when i did my analysis on this task. 
This is actually debatable. Even i am confused :)

Reply:
Can you create a ticket for it?

Reply:
We discussed only the approach/concept a bit but the review is related to code structure and making it cleaner. 

@JudeNiroshan's comment seems reasonable to me and looks valuable for future efforts because one needs to care for both cases for new implementation even they are running the same flow with just additional ifs (time complexity is O(1) so no performance issues). 

Also, I could not see any argument/argument related to the parallelism of syncMultiple, which is related (https://github.com/commercetools/commercetools-project-sync/pull/219#discussion_r544999238). 
If we could not find a way/algorithm to improve perf (as the current version), one or multiple might be the same algorithm.

Of course, the tradeoff of making the code cleaner is spending more time/effort...

Reply:
Also here we might decide to go with the syncAll approach for all type of sync, as all the algorithm could fit into the syncAll.